### ISSUE TITLE
Use DSA for signature verification. After the code generates the public and private key pem file, when reading the private key pem, the coredump

### ISSUE THREAD

**Issue by gttzsh** [openssl/openssl#19709]
This is my code. Please see if the usage is correct.

 ```
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <openssl/dsa.h>
 
int main() {
   static unsigned char seed[20] = {
      0xd5, 0x01, 0x4e, 0x4b, 0x60, 0xef, 0x2b, 0xa8,
      0xb6, 0x21, 0x1b, 0x40, 0x62, 0xba, 0x32, 0x24,
      0xe0, 0x42, 0x7d, 0xd3,
   };
   DSA* dsa = DSA_new();

   DSA_generate_parameters_ex(dsa, 1024, seed, 20, NULL, NULL, NULL);
   DSA_generate_key(dsa);
   DSA_print_fp(stdout, dsa, 0);
   printf("\n");

   // 生成pem
   makePriPem("./pri.pem", dsa);
   makePriPem("./pub.pem", dsa);
   return 0;
}

int makePriPem(const char *pub_key_fn, DSA* dsa) {
   BIO *out = NULL;
   out = BIO_new_file(pub_key_fn, "w");
   int ret = PEM_write_bio_DSAPrivateKey(out, dsa, NULL, NULL, 0, NULL, NULL);
   BIO_flush(out);
   BIO_free(out);
   DSA_print_fp(stdout, dsa, 0);
   return ret;
}

int makePubPem(const char *pub_key_fn, DSA* dsa) {
   BIO *out = NULL;   
   out = BIO_new_file(pub_key_fn, "w");
   int ret = PEM_write_bio_DSA_PUBKEY(out, dsa);
   BIO_flush(out);
   BIO_free(out);
   return ret;
}
 ```
```
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <openssl/dsa.h>
 
int main() {
   DSA* dsa = DSA_new();
   DSA_SIG *s = DSA_SIG_new();
   //printf("%ld---", sizeof(dsa));
   unsigned char input_string[23] = {49, 50, 51, 52, 53, 54, 55, 56, 57, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 48,78,89,90};
   unsigned char* sign_string;
   unsigned int sig_len = 0;
   unsigned int i;
   int r;
   dosin(input_string, 23, "./pri1.pem");
   return 0;
}

int dosin(const unsigned char *dgst, int dlen, const char *pri_key_fn) {
   DSA *prikey = DSA_new();
   DSA_SIG *s = DSA_SIG_new();
   BIO *priio = NULL;
   int ret = 0;

   priio = BIO_new_file(pri_key_fn, "rb");
   prikey = PEM_read_bio_DSAPrivateKey(priio, &prikey, NULL, NULL);
   if(prikey == NULL) {
      ret = -1;
      return ret;
   }
   DSA_print_fp(stdout, prikey, 0); //------coredump
   // s = DSA_do_sign(dgst, dlen, prikey);
   // if(s == NULL)
   // {
   //     ret = 0;
   //     DSA_free(prikey);
   //     return ret;
   // }
   //BIO_free(priio);
   //return (uint64_t)s;
   return 0;
}
```

**Comment by t8m** [comment]
I do not see anything apparently wrong with the code. You can avoid calling DSA_new() and just pass NULL instead of &prikey to the PEM_read_bio_DSAPrivateKey() but what you have there should also work.

What version of openssl are you using and could you run your program under debugger to get proper stack trace?

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
